### ISSUE TITLE
Usage instructions are out of date

### ISSUE THREAD

**Issue by zEasyIDK** [D-side/ftl_font#3]
I apologize to you, but I am bad at understanding and it is difficult to understand. Can you help me step by step?
1: i download and install ruby installer-2.4.1-1-x64
then ?

**Comment by D-side** [comment]
Hi!

Outside of <kbd>Clone or download</kbd> button here on GitHub replaced with just <kbd>Code</kbd>, the rest of the [usage instructions](https://github.com/D-side/ftl_font#usage) should still be working.

Also, I think the last project that was aiming to use this tool found that current version of FTL supports TTF font format and decided to **not** use this tool, as there's a lot of much more mature tooling around TTF already. They may be a better option for your project as well.

**Comment by zEasyIDK** [comment]
> Hi!
> 
> Outside of Clone or download button here on GitHub replaced with just Code, the rest of the [usage instructions](https://github.com/D-side/ftl_font#usage) should still be working.
> 
> Also, I think the last project that was aiming to use this tool found that current version of FTL supports TTF font format and decided to **not** use this tool, as there's a lot of much more mature tooling around TTF already. They may be a better option for your project as well.

Thank you for your response. Can you provide me with more information about the other tool?

**Comment by zEasyIDK** [comment]
> > Hi!
> > Outside of Clone or download button here on GitHub replaced with just Code, the rest of the [usage instructions](https://github.com/D-side/ftl_font#usage) should still be working.
> > Also, I think the last project that was aiming to use this tool found that current version of FTL supports TTF font format and decided to **not** use this tool, as there's a lot of much more mature tooling around TTF already. They may be a better option for your project as well.
> 
> Thank you for your response. Can you provide me with more information about the other tool?

Such as a link to download the tools or where to find the tools

**Comment by D-side** [comment]
Sadly, no. All I know is that it's standard TTF font building tools, nothing more specific than that.

FTL's modding community may be able to help. Last I checked, there's been one active channel on FTL's Discord and a forum section hosted by the game's developers.

**Comment by zEasyIDK** [comment]
> Sadly, no. All I know is that it's standard TTF font building tools, nothing more specific than that.
> 
> FTL's modding community may be able to help. Last I checked, there's been one active channel on FTL's Discord and a forum section hosted by the game's developers.

They did not know how to help me, so I have no choice but to use your tool. I downloaded the tool and extracted it, then what should I do?

**Comment by D-side** [comment]
Following instructions in the README is the intended path. If there's something missing, please let me know.

(edit) Just saw your ping on Discord about you not understanding the rest of the instructions. I don't hang around Discord that often, so I'd prefer to keep the discussion here. Anyways. Can you be more specific about what exactly you didn't understand?

**Comment by zEasyIDK** [comment]
> Following instructions in the README is the intended path. If there's something missing, please let me know.
> 
> (edit) Just saw your ping on Discord about you not understanding the rest of the instructions. I don't hang around Discord that often, so I'd prefer to keep the discussion here. Anyways. Can you be more specific about what exactly you didn't understand?

4. Install Bundler (`gem install bundler`) and use it to install dependencies (`bundle install`).

How do I install Bundler, through cmd, for example?

This is what appears when I do this using cmd
![image](https://github.com/D-side/ftl_font/assets/86298206/72f300ec-39d3-49e2-83d7-fb1274fac860)

**Comment by D-side** [comment]
That's the correct way and I have not encountered this condition when I was doing it back then.

Looking at the error message, this may have to do with Ruby not properly supporting your system language. However, if [this Stack Overflow answer](https://stackoverflow.com/a/22861206/2076787) is any indication, then running `chcp 1252` in the same cmd session beforehand should ~~fix~~ work around it. Give it a try.

**Comment by zEasyIDK** [comment]
> That's the correct way and I have not encountered this condition when I was doing it back then.
> 
> Looking at the error message, this may have to do with Ruby not properly supporting your system language. However, if [this Stack Overflow answer](https://stackoverflow.com/a/22861206/2076787) is any indication, then running `chcp 1252` in the same cmd session beforehand should ~fix~ work around it. Give it a try.

I installed Bundler, but the next command is bundle install. When I do this, I get this error, and when I open setup-ruby, the same error appears again.
![image](https://github.com/D-side/ftl_font/assets/86298206/71a43a72-53c3-4e05-8133-ac5dd62e2d13)

**Comment by D-side** [comment]
The actual error message is off-screen, but I think I see enough in the link it suggests.

A lot of time has passed, and installing **any** (as in, latest) Bundler might no longer be sufficient, as it might no longer be compatible with Ruby as old as 2.4.

Try *removing* Bundler:

```sh
gem uninstall bundler
```

…and instead installing the same version that I used back then (as indicated by `Gemfile.lock`):

```sh
gem install bundler -v 1.16.1
```

In fairness, this whole thing isn't complicated and might actually work fine with latest Ruby and Bundler as well, but I don't suppose you'd be comfortable testing it out.

**Comment by zEasyIDK** [comment]
> The actual error message is off-screen, but I think I see enough in the link it suggests.
> 
> A lot of time has passed, and installing **any** (as in, latest) Bundler might no longer be sufficient, as it might no longer be compatible with Ruby as old as 2.4.
> 
> Try _removing_ Bundler:
> 
> ```shell
> gem uninstall bundler
> ```
> 
> …and instead installing the same version that I used back then (as indicated by `Gemfile.lock`):
> 
> ```shell
> gem install bundler -v 1.16.1
> ```
> 
> In fairness, this whole thing isn't complicated and might actually work fine with latest Ruby and Bundler as well, but I don't suppose you'd be comfortable testing it out.

`C:\Users\Abdul Haq\Desktop\ftl_font-master>bundle install
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/shared_helpers.rb:29: warning: Pathname#untaint is deprecated and will be removed in Ruby 3.2.
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/shared_helpers.rb:118: warning: Pathname#untaint is deprecated and will be removed in Ruby 3.2.
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/shared_helpers.rb:118: warning: Pathname#untaint is deprecated and will be removed in Ruby 3.2.
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/shared_helpers.rb:35: warning: Pathname#untaint is deprecated and will be removed in Ruby 3.2.
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/shared_helpers.rb:35: warning: Pathname#untaint is deprecated and will be removed in Ruby 3.2.
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/shared_helpers.rb:35: warning: Pathname#untaint is deprecated and will be removed in Ruby 3.2.
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/shared_helpers.rb:44: warning: Pathname#untaint is deprecated and will be removed in Ruby 3.2.
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/shared_helpers.rb:118: warning: Pathname#untaint is deprecated and will be removed in Ruby 3.2.
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/shared_helpers.rb:118: warning: Pathname#untaint is deprecated and will be removed in Ruby 3.2.
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/shared_helpers.rb:35: warning: Pathname#untaint is deprecated and will be removed in Ruby 3.2.
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/shared_helpers.rb:44: warning: Pathname#untaint is deprecated and will be removed in Ruby 3.2.
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/shared_helpers.rb:118: warning: Pathname#untaint is deprecated and will be removed in Ruby 3.2.
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/shared_helpers.rb:118: warning: Pathname#untaint is deprecated and will be removed in Ruby 3.2.
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/shared_helpers.rb:118: warning: Pathname#untaint is deprecated and will be removed in Ruby 3.2.
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/shared_helpers.rb:118: warning: Pathname#untaint is deprecated and will be removed in Ruby 3.2.
Fetching source index from https://rubygems.org/
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/shared_helpers.rb:35: warning: Pathname#untaint is deprecated and will be removed in Ruby 3.2.
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/shared_helpers.rb:35: warning: Pathname#untaint is deprecated and will be removed in Ruby 3.2.
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/shared_helpers.rb:44: warning: Pathname#untaint is deprecated and will be removed in Ruby 3.2.
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/shared_helpers.rb:35: warning: Pathname#untaint is deprecated and will be removed in Ruby 3.2.
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/shared_helpers.rb:44: warning: Pathname#untaint is deprecated and will be removed in Ruby 3.2.
C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/shared_helpers.rb:35: warning: Pathname#untaint is deprecated and will be removed in Ruby 3.2.
--- ERROR REPORT TEMPLATE -------------------------------------------------------
# Error Report

## Questions

Please fill out answers to these questions, it'll help us figure out
why things are going wrong.

- **What did you do?**

  I ran the command `C:/Ruby30-x64/bin/bundle install`

- **What did you expect to happen?**

  I expected Bundler to...

- **What happened instead?**

  Instead, what happened was...

- **Have you tried any solutions posted on similar issues in our issue tracker, stack overflow, or google?**

  I tried...

- **Have you read our issues document, https://github.com/bundler/bundler/blob/master/doc/contributing/ISSUES.md?**

  ...

## Backtrace

```
NoMethodError: undefined method `inflate' for Gem:Module
  C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/fetcher.rb:103:in `fetch_spec'
  C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/remote_specification.rb:100:in `_remote_specification'
  C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/remote_specification.rb:106:in `method_missing'
  C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/installer.rb:216:in `block in ensure_specs_are_compatible!'
  C:/Ruby30-x64/lib/ruby/3.0.0/forwardable.rb:238:in `each'
  C:/Ruby30-x64/lib/ruby/3.0.0/forwardable.rb:238:in `each'
  C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/installer.rb:215:in `ensure_specs_are_compatible!'
  C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/installer.rb:84:in `block in run'
  C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/process_lock.rb:12:in `block in lock'
  C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/process_lock.rb:9:in `open'
  C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/process_lock.rb:9:in `lock'
  C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/installer.rb:72:in `run'
  C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/installer.rb:25:in `install'
  C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/cli/install.rb:65:in `run'
  C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/cli.rb:224:in `block in install'
  C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/settings.rb:136:in `temporary'
  C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/cli.rb:223:in `install'
  C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
  C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
  C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/vendor/thor/lib/thor.rb:387:in `dispatch'
  C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/cli.rb:27:in `dispatch'
  C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/vendor/thor/lib/thor/base.rb:466:in `start'
  C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/cli.rb:18:in `start'
  C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/exe/bundle:30:in `block in <top (required)>'
  C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/lib/bundler/friendly_errors.rb:122:in `with_friendly_errors'
  C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/bundler-1.16.1/exe/bundle:22:in `<top (required)>'
  C:/Ruby30-x64/bin/bundle:23:in `load'
  C:/Ruby30-x64/bin/bundle:23:in `<main>'
```

## Environment

```
Bundler       1.16.1
  Platforms   ruby, x64-mingw32
Ruby          3.0.0p0 (2020-12-25 revision 95aff214687a5e12c3eb57d056665741e734c188) [x64-mingw32]
  Full Path   C:/Ruby30-x64/bin/ruby.exe
  Config Dir  C:/ProgramData
RubyGems      3.2.3
  Gem Home    C:/Ruby30-x64/lib/ruby/gems/3.0.0
  Gem Path    C:/Ruby30-x64/lib/ruby/gems/3.0.0;C:/Users/Abdul Haq/.gem/ruby/3.0.0
  User Path   C:/Users/Abdul Haq/.gem/ruby/3.0.0
  Bin Dir     C:/Ruby30-x64/bin
OpenSSL
  Compiled    OpenSSL 1.1.1i  8 Dec 2020
  Loaded      OpenSSL 1.1.1i  8 Dec 2020
  Cert File   C:/Ruby30-x64/ssl/cert.pem
  Cert Dir    C:/Ruby30-x64/ssl/certs
Tools
  Git         not installed
  RVM         not installed
  rbenv       not installed
  chruby      not installed
```

## Bundler Build Metadata

```
Built At          2017-12-21
Git SHA           0034ef341
Released Version  true
```

## Gemfile

### Gemfile

```ruby
source "https://rubygems.org"

gem "chunky_png"
gem "bit-struct"

gem "pry"
gem "pry-doc"
```

### Gemfile.lock

```
GEM
  remote: https://rubygems.org/
  specs:
    bit-struct (0.16)
    chunky_png (1.3.8)
    coderay (1.1.2)
    method_source (0.9.0)
    pry (0.11.3)
      coderay (~> 1.1.0)
      method_source (~> 0.9.0)
    pry-doc (0.11.1)
      pry (~> 0.9)
      yard (~> 0.9)
    yard (0.9.12)

PLATFORMS
  ruby
  x64-mingw32

DEPENDENCIES
  bit-struct
  chunky_png
  pry
  pry-doc

BUNDLED WITH
   1.16.1
```

--- TEMPLATE END ----------------------------------------------------------------

Unfortunately, an unexpected error occurred, and Bundler cannot continue.

First, try this link to see if there are any existing issue reports for this error:
https://github.com/bundler/bundler/search?q=undefined+method+%60inflate%27+for+Gem+Module&type=Issues

If there aren't any reports for this error yet, please create copy and paste the report template above into a new issue. Don't forget to anonymize any private data! The new issue form is located at:
https://github.com/bundler/bundler/issues/new

C:\Users\Abdul Haq\Desktop\ftl_font-master>gem uninstall bundler
Gem bundler-2.2.3 cannot be uninstalled because it is a default gem
Successfully uninstalled bundler-1.16.1

C:\Users\Abdul Haq\Desktop\ftl_font-master>gem uninstall bundler
Gem 'bundler' is not installed

C:\Users\Abdul Haq\Desktop\ftl_font-master>gem uninstall bundler -v 2.2.3
Gem bundler-2.2.3 cannot be uninstalled because it is a default gem

C:\Users\Abdul Haq\Desktop\ftl_font-master>

C:\Users\Abdul Haq\Desktop\ftl_font-master>C:\Users\Abdul Haq\Desktop\ftl_font-master>gem uninstall bundler -v 2.2.3
'C:\Users\Abdul' is not recognized as an internal or external command,
operable program or batch file.`

**Comment by zEasyIDK** [comment]
i got all this from cmd

**Comment by D-side** [comment]
Ah.

> Ruby          3.0.0p0

So you're **not** running Ruby 2.4.

2.4 is in the archives: https://rubyinstaller.org/downloads/archives/ — I believe the newest is `2.4.10-1`

**Comment by D-side** [comment]
I've updated the instructions and the setup script so that others hopefully won't have to go through the troubles you had here.

Did it work? No more issues? Can I close this now?

**Comment by zEasyIDK** [comment]
> I've updated the instructions and the setup script so that others hopefully won't have to go through the troubles you had here.
> 
> Did it work? No more issues? Can I close this now?

Did he succeed or fail?
![image](https://github.com/D-side/ftl_font/assets/86298206/737d4954-3c6f-4ee1-bb43-5bed76892f11)

**Comment by D-side** [comment]
You seem to have installed the right Bundler before, given the output after the error, so this looks like success, yes.

I fixed the script so that it installs the right Bundler too, feel free to replicate this change on your end: https://github.com/D-side/ftl_font/pull/4/commits/faa5a157e3175ed4a2ceb0367ec9ed3307f400d2

**Comment by zEasyIDK** [comment]
> You seem to have installed the right Bundler before, given the output after the error, so this looks like success, yes.
> 
> I fixed the script so that it installs the right Bundler too, feel free to replicate this change on your end: [faa5a15](https://github.com/D-side/ftl_font/commit/faa5a157e3175ed4a2ceb0367ec9ed3307f400d2)

It's not practical to talk here, can we talk on Discord?

**Comment by D-side** [comment]
I don't normally keep it running and you may have to nudge me here if I stop responding there.
But that aside — yeah, sure.

**Comment by zEasyIDK** [comment]
> You seem to have installed the right Bundler before, given the output after the error, so this looks like success, yes.
> 
> I fixed the script so that it installs the right Bundler too, feel free to replicate this change on your end: [faa5a15](https://github.com/D-side/ftl_font/commit/faa5a157e3175ed4a2ceb0367ec9ed3307f400d2)

Well what do I do now?
![Uploading image.png…]()

**Comment by D-side** [comment]
_📌 For transparency for the other readers of this thread_

We continued this conversation on Discord and it looks like #4 has indeed fixed all the issues pertaining to setting up this tool years later it was built. We got through dismantling and reassembling the examples in the repository, and both procedures seemed to work with no issues.

We also briefly discussed the language barrier which I filed separately as #5.
Aging of the setup instructions seems to be handled, so I'm closing this one.
